src/components/Attachment/Gallery.js

```diff
@@ -1,294 +1,240 @@
-        image_url: PropTypes.string,
-        thumb_url: PropTypes.string,
-      }),
-    ),
-    onLongPress: PropTypes.func,
-    /**
-     * Provide any additional props for child `TouchableOpacity`.
-     * Please check docs for TouchableOpacity for supported props - https://reactnative.dev/docs/touchableopacity#props
-     */
-    additionalTouchableProps: PropTypes.object,
-    alignment: PropTypes.string,
-  };
+  const [viewerModalImageIndex, setViewerModalImageIndex] = useState(0);
+  const [viewerModalOpen, setViewerModalOpen] = useState(false);
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      viewerModalOpen: false,
-      viewerModalImageIndex: 0,
-    };
-  }
+  if (!images || !images.length) return null;
 
-  render() {
-    const { t, additionalTouchableProps } = this.props;
-    if (!this.props.images || this.props.images.length === 0) return null;
-
-    const images = [...this.props.images].map((i) => ({
-      url: makeImageCompatibleUrl(i.image_url || i.thumb_url),
-    }));
-
-    if (images.length === 1) {
-      return (
-        <React.Fragment>
-          <Single
-            onPress={() => {
-              this.setState({ viewerModalOpen: true });
-            }}
-            onLongPress={() => {
-              this.props.onLongPress();
-            }}
-            alignment={this.props.alignment}
-            {...additionalTouchableProps}
-          >
-            <Image
-              style={{
-                width: 100 + '%',
-                height: 100 + '%',
-              }}
-              resizeMode='cover'
-              source={{ uri: images[0].url }}
-            />
-          </Single>
-          <Modal
-            visible={this.state.viewerModalOpen}
-            transparent={true}
-            onRequestClose={() => {
-              this.setState({ viewerModalOpen: false });
-            }}
-          >
-            <SafeAreaView style={{ flex: 1, backgroundColor: 'transparent' }}>
-              <ImageViewer
-                imageUrls={images}
-                // TODO: We don't have 'save image' functionality.
-                // Until we do, lets disable this feature. saveToLocalByLongPress prop basically
-                // opens up popup menu to with an option "Save to the album", which basically does nothing.
-                saveToLocalByLongPress={false}
-                onCancel={() => {
-                  this.setState({ viewerModalOpen: false });
-                }}
-                enableSwipeDown
-                renderHeader={() => (
-                  <GalleryHeader
-                    handleDismiss={() => {
-                      this.setState({ viewerModalOpen: false });
-                    }}
-                  />
-                )}
-              />
-            </SafeAreaView>
-          </Modal>
-        </React.Fragment>
-      );
-    }
+  const galleryImages = [...images].map((i) => ({
+    url: makeImageCompatibleUrl(i.image_url || i.thumb_url),
```

Comment:
`i` means `index` to me
```suggestion
  const galleryImages = [...images].map((image) => ({
    url: makeImageCompatibleUrl(image.image_url || image.thumb_url),
```